src/server.c

```diff
@@ -1453,8 +1453,8 @@ int serverCron(struct aeEventLoop *eventLoop, long long id, void *clientData) {
             server.rdb_bgsave_scheduled = 0;
     }
 
-    run_with_period(100) {
-        if (moduleCount()) modulesCron();
+    if (moduleCount()) {
+        run_with_period(100) modulesCron();
```

Comment:
Module count is not heavier to compute than run with period? E.g. one memory access vs zero...

Reply:
Nevermind